contracts/data/BinaryHeap.sol

```diff
@@ -0,0 +1,320 @@
+    }
+
+    function length(Bytes32Heap storage heap) internal view returns (uint256) {
+        return _length(heap._inner);
+    }
+
+    function length(AddressHeap storage heap) internal view returns (uint256) {
+        return _length(heap._inner);
+    }
+
+    function length(UintHeap storage heap) internal view returns (uint256) {
+        return _length(heap._inner);
+    }
+
+    function root(Bytes32Heap storage heap) internal view returns (bytes32) {
+        return _root(heap._inner);
+    }
+
+    function root(AddressHeap storage heap) internal view returns (address) {
+        return address(uint160(uint256(_root(heap._inner))));
+    }
+
+    function root(UintHeap storage heap) internal view returns (uint256) {
+        return uint256(_root(heap._inner));
+    }
+
+    function add(Bytes32Heap storage heap, bytes32 value)
+        internal
+        returns (bool)
+    {
+        return _add(heap._inner, value);
+    }
+
+    function add(AddressHeap storage heap, address value)
+        internal
+        returns (bool)
+    {
+        return _add(heap._inner, bytes32(uint256(uint160(value))));
+    }
+
+    function add(UintHeap storage heap, uint256 value) internal returns (bool) {
+        return _add(heap._inner, bytes32(value));
+    }
+
+    function remove(Bytes32Heap storage heap, bytes32 value)
+        internal
+        returns (bool)
+    {
+        return _remove(heap._inner, value);
+    }
+
+    function remove(AddressHeap storage heap, address value)
+        internal
+        returns (bool)
+    {
+        return _remove(heap._inner, bytes32(uint256(uint160(value))));
+    }
+
+    function remove(UintHeap storage heap, uint256 value)
+        internal
+        returns (bool)
+    {
+        return _remove(heap._inner, bytes32(value));
+    }
+
+    function toArray(Bytes32Heap storage heap)
+        internal
+        view
+        returns (bytes32[] memory)
+    {
+        uint256 len = _length(heap._inner);
+        bytes32[] memory arr = new bytes32[](len);
+        for (uint256 index = 0; index < len; index++) {
+            arr[index] = _at(heap._inner, index);
+        }
+        return arr;
+    }
+
+    function toArray(AddressHeap storage heap)
+        internal
+        view
+        returns (address[] memory)
+    {
+        uint256 len = _length(heap._inner);
+        address[] memory arr = new address[](len);
+        for (uint256 index = 0; index < len; index++) {
+            arr[index] = address(uint160(uint256(_at(heap._inner, index))));
+        }
+        return arr;
+    }
+
+    function toArray(UintHeap storage heap)
+        internal
+        view
+        returns (uint256[] memory)
+    {
+        uint256 len = _length(heap._inner);
+        uint256[] memory arr = new uint256[](len);
+        for (uint256 index = 0; index < len; index++) {
+            arr[index] = uint256(_at(heap._inner, index));
+        }
+        return arr;
+    }
```

Comment:
The `for` loops in each of the `toArray` functions can be marked `unchecked`.

Reply:
Also, instead of casting the value returned by the low-level `_at` function, why not use the higher-level `at` functions that already return the desired type?

Reply:
Great suggestion! https://github.com/solidstate-network/solidstate-solidity/pull/134/commits/2f4fde36bcf8cd3220fd89abdf6972aef5263be4